### 01. Groupby입니다 ^^ - 조금 어렵습니다 ㅠㅠ

뉴욕 airBnB : https://www.kaggle.com/ptoscano230382/air-bnb-ny-2019 

In [1]:
# Q1. 데이터를 로드한 뒤 상위 5개 컬럼을 출력해보자.

In [2]:
import pandas as pd
import numpy as np

In [3]:
air_df = pd.read_csv('C:/Users/sjy/dataframes/archive_data/AB_NYC_2019.csv')
air_df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [4]:
# Q2. 각 host_name의 빈도수를 구하고 host_name으로 정렬하여 상위 5개를 출력해보자. => .size 함수 이용
# 1) 보통 빈도수를 구하는
# Groupby문제는 .size()함수가 활용되기도 합니다.
# df.groupby(['컬럼명']).size()

In [73]:
air_df.groupby(by='host_name').size().head()

host_name
#NAME?                      1
'Cil                        1
(Ari) HENRY LEE             1
(Email hidden by Airbnb)    6
(Mary) Haiy                 1
dtype: int64

In [6]:
# Q3. 데이터의 각 host_name의 빈도수를 구하고, 
# 빈도수 기준 내림차순 정렬한 데이터 프레임을 생성해보자.
# 빈도수 컬럼의 컬럼명은 counts로 변경해보자.(host_name은 index입니당 ^^;)

In [103]:
# 1번째 방법-value_counts()를 활용하는 방법
new_df = pd.DataFrame(air_df.host_name.value_counts()).rename(columns={'host_name': 'counts'})
new_df

counts
Michael               417
David                 403
Sonder (NYC)          327
John                  294
Alex                  279
...                   ...
Rhonycs                 1
Brandy-Courtney         1
Shanthony               1
Aurore And Jamila       1
Ilgar & Aysel           1

[11452 rows x 1 columns]

In [105]:
# 2번째 방법 - Groupby() :: index 명을 지우지 않는다... 
air_df.groupby(by='host_name').size().to_frame().rename(columns={0:'counts'}).sort_values(by='counts',ascending=False)

counts
host_name           
Michael          417
David            403
Sonder (NYC)     327
John             294
Alex             279
...              ...
Jerbean            1
Jerald             1
Jeonghoon          1
Jeny               1
현선                 1

[11452 rows x 1 columns]

||counts|
|---|---|
|host_name||
|Michael|417|
|David|403|
|Sonder (NYC)|327|
|John|294|
|Alex|279|

In [7]:
# Q4. neighbourhood_group값에 따른 neighbourhood컬럼 값의 갯수를 구해보세요.

In [109]:
air_df.groupby(['neighbourhood_group'])['neighbourhood'].value_counts()

neighbourhood_group  neighbourhood
Bronx                Kingsbridge      70
                     Fordham          63
                     Longwood         62
                     Mott Haven       60
                     Concourse        50
                                      ..
Staten Island        New Dorp          1
                     Richmondtown      1
                     Rossville         1
                     Willowbrook       1
                     Woodrow           1
Name: neighbourhood, Length: 221, dtype: int64

|neighbourhood_group|neighbourhood|size|
|---|---|---|
|Bronx|Allerton|42|
|Bronx|Baychester|7|
|Bronx|Belmont|24|
|Bronx|Bronxdale|19|


In [8]:
# Q5. neighbourhood_group값에 따른 reviews_per_month 값의 평균, 분산, 최대, 최소값을 구해보자- agg함수 활용

In [9]:
# Q6. neighbourhood_group값과 neighbourhood값에 따른 price의 평균을 구해보자.

### 02. Apply_lambda 

카드이용데이터 : https://www.kaggle.com/sakshigoyal7/credit-card-customers 

In [10]:
# Q1. 데이터를 로드하고, 데이터의 정보를 확인해보세요.

In [67]:
# Q2. Income_Category의 카테고리를 다음과 같이 변경하여 newIncome 컬럼에 넣어주세요 ^^ 
# map함수를 활용하셔도, apply를 활용하셔도 상관없습니다 ^^
Unknown : N
Less than $40K : a
$40K - $60K : b
$60K - $80K : c
$80K - $120K : d
$120K +’ : e

SyntaxError: invalid syntax (3827884766.py, line 4)

In [ ]:
# Q3. Customer_Age의 값을 이용하여 나이 구간을 Age_cat이라는 새 컬럼을 만들어주세요. (0~9:0, 10~19:10,
# 20~29:20...) 각 구간의 빈도수를 출력하라.

In [ ]:
# Q4. Education_Level의 값에서 Graduate이 포함된 경우에는 1 그렇지 않다면 0으로 변경하여
# new_edu_level 컬럼을 정의하고 빈도수를 출력해보세요.

In [66]:
# Q5. Marital_Status 컬럼값이 Married이고 Card_Category 컬럼의 값이 Platinum인 경우 1 그렇지 
# 않은 경우에는 모두 0으로 정의해주세요...

# 그리고 그 결과를 new_status 컬럼에 넣어주세요.